### ISSUE TITLE
"no-unused-para-var" only check the last parameter

### ISSUE THREAD

**Issue by enobufs** [eslint/eslint#4602]
I am not convinced by the conclusion with the previous issue, #4292. Let me open another issue here.

Just to recap, following code does not detect unused vars, arg1.

```js
var linter = require('eslint').linter;

var result = linter.verify([
    'function doSomething(arg1, arg2) {',
    '    return 1 + arg2',
    '}',
    'void(doSomething());'
].join('\n'), {
    rules: {
        "no-unused-vars": 2
    }
});

console.log(result);
// []
```

The previous issue concluded that it is an expected behavior with this justification:
>  If you are using 4th parameter, you need previous 3 parameters to get the right value. Even if they are unused, they serve a purpose.

The function in the above example obviously has a bug, and "no-unused-vars" is not capable of detecting it. I would definitely expect eslint to detect unsed var arg1.

I am using eslint v1.10.3

**Comment by eslintbot** [comment]
@enobufs Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
You can catch this scenario with `no-unused-vars: [2, { "args": "all" }]`.

**Comment by enobufs** [comment]
Ohh...  I did not notice that option.  Thank you!
